src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -767,6 +768,9 @@ protected function getContainerBuilder()
         $container = new ContainerBuilder();
         $container->getParameterBag()->add($this->getKernelParameters());
 
+        if ($this instanceof CompilerPassInterface) {
+            $container->setKernelPass($this);
```

Comment:
Why do we need to set the kernel pass in the container builder (which is a bit weird to me) rather than simply adding it through `$container->getCompilerPassConfig()->addPass(...)` or `$container->addCompilerPass(...)`? Is processing the pass right after the extensions (in `ExtensionCompilerPass`) the best option? I think it probably is, but I'd like to get the full reasoning behind :)

Reply:
Good question :)
The reason is as you said that this pass should be run last after bundle passes, and before optimization passes.

Reply:
Actually, I now changed the implementation so that no new method is needed. Using priorities should be good enough and better in fact. WDYT?

Reply:
Much better to me 👍
It can be a small BC break for people already relying on priorities though. So should we mention this somewhere?
Also the CHANGELOG.md entry is missing :)